src/components/AddItem.js

```diff
@@ -5,9 +5,30 @@ import HowSoonOptions from './HowSoonOptions';
 import '../css/components/AddItemsForm.css';
 import AddItemInput from './AddItemInput';
 import { getToken } from '../lib/TokenService';
+import { existingName } from '../lib/helper';
 
 const AddItem = () => {
   let [inputValue, setInputValue] = useState('');
+  const [message, setMessage] = useState('');
+  const [messageType, setMessageType] = useState('');
+
+  const displayMessage = (message, type) => {
+    setMessage(message);
+    setMessageType(type);
+    hideMessage();
+  };
+
+  const hideMessage = () => {
+    setTimeout(() => {
```

Comment:
I like this approach a lot! You have to be careful with setTimeout, it's tricky and it gets messy super fast.

Try to add 2 elements to the list fast and you will see that by adding the second item, the first message is hidden and you don't have confirmation for the second message.

The problem is normally that by calling setTimeout in that way, it creates multiple independent timers, and they start to stack. There are multiple approaches to handle this situation, with vanilla JS the easiest way is to have only one instance of setTimeout by storing the identifier it returns and call clearTimeout if another event comes before the 5000ms https://www.w3schools.com/jsref/met_win_cleartimeout.asp

This is non blocking, good work!

Reply:
That's an interesting point
thanks for spotting it out